scripts/restore

```diff
@@ -0,0 +1,183 @@
+focus_password=$(ynh_app_setting_get --app=$app --key=focus_password)
+
+port=$(ynh_app_setting_get --app=$app --key=port)
+port_videobridge=$(ynh_app_setting_get --app=$app --key=port_videobridge)
+port_component=$(ynh_app_setting_get --app=$app --key=port_component)
+
+#=================================================
+# CHECK IF THE APP CAN BE RESTORED
+#=================================================
+ynh_script_progression --message="Validating restoration parameters..." --time --weight=1
+
+ynh_webpath_available --domain=$domain --path_url=$path_url \
+	|| ynh_die --message="Path not available: ${domain}${path_url}"
+test ! -d $final_path \
+	|| ynh_die --message="There is already a directory: $final_path "
+
+#=================================================
+# STANDARD RESTORATION STEPS
+#=================================================
+# RESTORE THE NGINX CONFIGURATION
+#=================================================
+
+ynh_restore_file --origin_path="/etc/nginx/conf.d/$domain.d/$app.conf"
+
+#=================================================
+# RESTORE THE APP MAIN DIR
+#=================================================
+ynh_script_progression --message="Restoring the app main directory..." --time --weight=1
+
+ynh_restore_file --origin_path="$final_path"
+
+#=================================================
+# RECREATE THE DEDICATED USER
+#=================================================
+ynh_script_progression --message="Recreating the dedicated system user..." --time --weight=1
+
+# Create the dedicated user (if not existing)
+ynh_system_user_create --username=$app --home_dir=$final_path
+
+#=================================================
+# RESTORE USER RIGHTS
+#=================================================
+
+# Restore permissions on app files
+chown -R root: $final_path
+
+#=================================================
+# SPECIFIC RESTORATION
+#=================================================
+# REINSTALL DEPENDENCIES
+#=================================================
+ynh_script_progression --message="Reinstalling dependencies..." --time --weight=1
+
+# Define and install dependencies
+ynh_install_app_dependencies $pkg_dependencies
+
+ynh_install_nodejs --nodejs_version=10
+
+#=================================================
+# CONFIGURE FIREWALL
+#=================================================
+ynh_script_progression --message="Configuring firewall..." --time --weight=1
+
+# Open this port
+ynh_exec_warn_less yunohost firewall allow --no-upnp TCP $port
+ynh_exec_warn_less yunohost firewall allow --no-upnp UDP $port_videobridge
+ynh_exec_warn_less yunohost firewall allow --no-upnp TCP $port_component
+
+#=================================================
+# CONFIGURE METRONOME
+#=================================================
+ynh_script_progression --message="Configuring metronome..." --time --weight=1
+
+# Create additional domains
+if [[ $domain == *"domain.tld"* ]]; then
```

Comment:
O.o `domain.tld`??? Some explanation please.

Reply:
Ok, I get it! This is for package check! It's not possible now with the 3.6 to do it?

Reply:
Difficult decision, the point is:  Jitsi need additional subdomains to work. https://github.com/YunoHost-Apps/jitsi_ynh/pull/4/files#diff-04c6e90faac2675aa89e2176d2eec7d8R19

So there is two options:
1. Let the YunoHost admin: create the subdomains in his DNS registrar, create the subdomains and install certificate from Let's Encrypt
2. Let the YunoHost admin create the subdomains in his DNS registrar and the install script will create subdomains and request certificate.

In both cases, we have to make a workaround for package_check tests:
1. We have to create the subdomains during install during package_check because they are requested during install
2. We have to not request Let's Encrypt certificate because domain.tld doesn't exist

During the packaging of the application, I meet a limitation: you can't ask more than 5 times a week a certificate from Let's Encrypt for a domain name....

Therefore, when thinking of YunoHost users installing Jitsi_YNH, if their installation fails several times, maybe they will need to wait one week before retrying to install it.... I let you imagine the number of support request for that situation..... And the headaches for me and the support team...

It's why I decided to implement option 1. => having the user creating the subdomains and requesting the certificates manually and then installing jitsi_ynh.

It's whey there is that ugly workaround, looking if we are in a package_check, and if we are, it creates the requested subdomains...


When you say: `It's not possible now with the 3.6 to do it?` I don't understand what in 3.6 can be helpfull in that situation, can you explain? Maybe I missed something..

When you say

Reply:
Just for your information: As I understand it there were changes with the 5 domains per week Limit.
see: https://letsencrypt.org/docs/rate-limits/

In my understanding there are 50 domains per week per registered domain.
And the 5 per week limit: "Renewals are treated specially: they don’t count against your Certificates per Registered Domain limit, but they are subject to a **Duplicate** Certificate limit of 5 per week."

Does that change the decision to not create the additional domains?

Reply:
The point is the duplicate certificate limit of 5 per week.
If the install fails for several reasons, after 5 attempts, you would need to wait a week before retrying to install...